ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V114__activate_string_interning_for_parties.sql

```diff
@@ -0,0 +1,317 @@
+--  Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+--  SPDX-License-Identifier: Apache-2.0
+
+SELECT 'Party interning data migration - Preparation...';
+
+-- add temporary index to aid string lookups as migrating data
+CREATE UNIQUE INDEX string_interning_external_string_temp_idx ON string_interning USING btree (external_string);
+
+-- drop participant_events view (enables column type changes)
+DROP VIEW participant_events;
+
+-- add temporary migration function
+CREATE FUNCTION internalize_string_array(prefix TEXT, string_array TEXT[])
+RETURNS INTEGER[]
+LANGUAGE plpgsql
+AS
+$$
+DECLARE
+   result INTEGER[];
+BEGIN
+   EXECUTE
+   'SELECT CASE' ||
+   '  WHEN $2 IS NULL THEN NULL' ||
+   '  WHEN cardinality($2) = 0 THEN ARRAY[]::INTEGER[]' ||
+   '  ELSE (' ||
+   '    SELECT array_agg(string_interning.internal_id)' ||
+   '      FROM' ||
+   '        unnest($2) AS original(string_elem),' ||
```

Comment:
comma is equivalent to a CROSS JOIN that is only later whittled down with a WHERE statement on the result table. I would suggest replacing with a INNER JOIN.
I don't now to what extent Postgres is smart enough to optimize in this manner behind the scenes anyway

Reply:
We use cross joins everywhere, and in all cases it could be transformed to an inner join (so we never need a true cross join).
My working assumption is that it does not really matter which you choose from planner point of view (at least for PostgreSQL)
I checked the plans for the cross vs inner join in that particular case, and it is identical:
```
       explain SELECT array_agg(string_interning.internal_id)
         FROM
           unnest(array['a', 'b', 'c', 'ACSinvalidLedgerId-alpha-b894853580a6-party-0', 'ACSemptyResponse-alpha-b894853580a6-party-0']) AS original(string_elem)
           inner join string_interning on ('p|' || original.string_elem = string_interning.external_string);
```
```
                                          QUERY PLAN                                          
----------------------------------------------------------------------------------------------
 Aggregate  (cost=23.03..23.04 rows=1 width=32)
   ->  Hash Join  (cost=0.12..23.02 rows=5 width=4)
         Hash Cond: (string_interning.external_string = ('p|'::text || original.string_elem))
         ->  Seq Scan on string_interning  (cost=0.00..18.90 rows=790 width=68)
         ->  Hash  (cost=0.05..0.05 rows=5 width=32)
               ->  Function Scan on unnest original  (cost=0.00..0.05 rows=5 width=32)
```
I think we can leave it as it is.

Reply:
:+1: as comma join also happens to be more redable